config/config_test.go

```diff
@@ -17,55 +17,116 @@ import (
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
+	vaultconfig "github.com/hashicorp/vault/command/agent/config"
 )
 
 func TestLoadConfig(t *testing.T) {
 
 	cases := []struct {
-		name string
-		file string
-		err  string
+		name         string
+		file         string
+		err          string
+		expectConfig *vaultconfig.Config
 	}{
 		{
 			"file-doesnt-exist",
 			"testdata/nonexistent.hcl",
 			"stat testdata/nonexistent.hcl: no such file or directory",
+			nil,
 		},
 		{
 			"provided-directory",
 			"testdata",
 			"location is a directory, not a file",
+			nil,
 		},
 		{
 			"empty-file",
 			"testdata/empty-file.hcl",
 			"no 'auto_auth' block found in configuration file",
+			nil,
 		},
 		{
 			"no-method",
 			"testdata/no-method.hcl",
 			"error parsing 'auto_auth': error parsing 'method': one and only one \"method\" block is required",
+			nil,
 		},
 		{
 			"no-sinks",
 			"testdata/no-sinks.hcl",
 			"",
+			&vaultconfig.Config{
+				AutoAuth: &vaultconfig.AutoAuth{
+					Method: &vaultconfig.Method{
+						Type:      "approle",
+						MountPath: "auth/approle",
+						Config: map[string]interface{}{
+							"role_id_file_path":   "/tmp/role-id",
+							"secret":              "secret/docker/creds",
+							"secret_id_file_path": "/tmp/secret-id",
+						},
+					},
+				},
+			},
 		},
 		{
 			"no-mount-path",
 			"testdata/no-mount-path.hcl",
 			"",
+			&vaultconfig.Config{
+				AutoAuth: &vaultconfig.AutoAuth{
+					Method: &vaultconfig.Method{
+						Type:      "aws",
+						MountPath: "auth/aws",
```

Comment:
The test name is `no-mount-path` but you are specifying a mount path here. Is this intentional, or am I missing something?

Reply:
Yup, that's by design. This test checks that if no mount path is in the config, then `config.LoadConfig` sets it to the default value. See [this](https://github.com/hashicorp/vault/blob/master/command/agent/config/config.go#L329-L330) block.